src/states_screens/dialogs/ghost_replay_info_dialog.cpp

```diff
@@ -59,17 +96,98 @@ GhostReplayInfoDialog::GhostReplayInfoDialog(unsigned int replay_id)
     }
 
     m_record_widget->setState(false);
-    m_watch_widget->setState(false);
+    m_watch_widget->setState(m_compare_ghost);
+    m_compare_widget->setState(m_compare_ghost);
+
+    if (m_compare_ghost)
+    {
+        m_watch_only = true;
+        m_record_race = false;
+        m_record_widget->setState(false);
+        m_record_widget->setVisible(!m_watch_only);
+        getWidget<LabelWidget>("record-race-text")->setVisible(!m_watch_only);
+    }
+
+    // Display this checkbox only if there is another replay file to compare with
+    getWidget<LabelWidget>("compare-ghost-text")->setVisible(m_compare_ghost);
+    m_compare_widget->setVisible(m_compare_ghost);
+
 
     m_action_widget->setFocusForPlayer(PLAYER_ID_GAME_MASTER);
     m_action_widget->select("start", PLAYER_ID_GAME_MASTER);
 }   // GhostReplayInfoDialog
-
 // -----------------------------------------------------------------------------
 GhostReplayInfoDialog::~GhostReplayInfoDialog()
 {
 }   // ~GhostReplayInfoDialog
 
+// -----------------------------------------------------------------------------
+void GhostReplayInfoDialog::updateReplayDisplayedInfo()
+{
+    m_replay_info_widget->clear();
+
+    int n=2;
+    if (m_compare_ghost)
+        n++;
+
+    bool is_linear = GhostReplaySelection::getInstance()->isActiveModeLinear();
+
+    for (int i=0; i<n; i++)
+    {
+        int id;
+
+        if (i<=1)
```

Comment:
It's unclear to me why replays 0-1 are special?

Reply:
Actually, there are 3 things this function push to the list : the header, the data for the current selected replay (the one the player whose selection triggered the modal dialog), and the replay used for comparison.

As the code for the first replay and for the second replay is mostly identical, I do a if/else to get the id, get the ReplayData from it, and then use it. The header don't need that ReplayData, but in this setup I need to feed some id value to it.

It would probably clearer if I moved the header code out of the loop.